### ISSUE TITLE
[compare] difference images have several issues if either of the images compared is .heic

### ISSUE THREAD

**Issue by shanemcquillan** [ImageMagick/ImageMagick#1277]
### Description
When using the `compare` command line tool, the difference images have several issues if either of the images compared is .heic. When I compare the linked images `image.heic` and `image.png` (they are the same size and visibly the same image; the png was generated from the heic version using macOS's Preview app) I get a huge difference score (1.21928e+07), and the resulting .png difference image (`diff_no_fuzz.png`, linked below) is neon green throughout with the actual image faintly visible in the background.

By specifying a fuzz factor of 40% (the difference is still huge with any lower fuzz factor, which is odd given that they're visibly identical. This may be its own issue) I can get the difference score down to 36594 pixels. In this case the differences are still highlighted in neon green, but the similarities are highlighted in purple (`diff_fuzz_40.png`, linked below). This coloring is only used when either image is .heic.

I attempted to specify the -highlight-color and -lowlight-color flags, but neither had any effect.

Additional notes:
- These issues also reproduce when I try formats other than .png for the output comparison image
- I have seen similarly unusual issues when both images are .heic.
- .heic does not work as an output format at this point. I believe this is because there is currently no encoder support for heic images in ImageMagick.

### Steps to Reproduce
```
compare -metric ae image.heic image.png diff_no_fuzz.png
compare -metric ae -fuzz 40% image.heic image.png diff_fuzz_40.png
```

### System Configuration

**- ImageMagick version:** 
```
compare --version
Version: ImageMagick 7.0.8-9 Q16 x86_64 2018-08-11 https://www.imagemagick.org
Copyright: © 1999-2018 ImageMagick Studio LLC
License: https://www.imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules
Delegates (built-in): bzlib cairo fontconfig freetype heic jng jp2 jpeg lcms ltdl lzma openexr png rsvg tiff xml zlib
```
**- Environment (Operating system, version and so on):** 
macOS High Sierra 10.13.2

**- Additional information:** 
libde265 version: 1.0.3
libheif version: 1.3.2

**Images (too large to attach here):**
image.png: https://drive.google.com/file/d/1HoBMLnnU9EFKlYneclHUc8seNZ6gv7kq/view?usp=sharing
image.heic: https://drive.google.com/file/d/1140M0r6YoOxxDzyY7Ee_aeTjiu3Q2UzA/view?usp=sharing
diff_no_fuzz.png: https://drive.google.com/file/d/1mBh08YalzPNyI3PU-4m09dF5uinrhrH0/view?usp=sharing
diff_fuzz_40.png: https://drive.google.com/file/d/1RtLG5D_wYmGXVIwKqcAFnWSQsEmkJyUj/view?usp=sharing

**Comment by dlemstra** [comment]
Heic images have the YCbCr colorspace what happens when you change them to sRGB first?

**Comment by shanemcquillan** [comment]
Good call, I made the wrong assumption that the colorspace was normalized automatically. The comparison image works well when I use sRGB as the colorspace during the comparison. However, interestingly the comparison image is still showing the issues I described above if I specify the colorspace as YCbCr (though the comparison results make more sense). Is that expected? Or is there an issue there?

I'm also slightly concerned that the switch in colorspace will affect the comparison results due to the loss in precision during conversion. Are there any recommendations to minimize this loss in precision? I'd imagine selecting a colorspace with a wider gamut makes most sense, as conversion to one with with a narrower gamut will always lead to loss in precision.
